lib/beaker-hostgenerator/generator.rb

```diff
@@ -32,43 +37,38 @@ def generate(tokens, options)
           next
         end
 
-        node_info = __parse_node_info_token(token)
+        node_info = parse_node_info_token(token)
+
+        # Build node host name
+        platform = "#{ostype}-#{node_info['bits']}"
+        host_name = "#{platform}-#{nodeid[ostype]}"
 
+        node_info['platform'] = platform
         node_info['ostype'] = ostype
         node_info['nodeid'] = nodeid[ostype]
 
-        host_config = base_host_config()
-
-        [:pe_dir, :pe_ver, :pe_upgrade_dir, :pe_upgrade_ver].each do |option|
-          if options[option]
-            host_config[option.to_s] = options[option]
-          end
-        end
+        host_config = base_host_config(options)
 
+        # Delegate to the hypervisor
         hypervisor = BeakerHostGenerator::Hypervisor.create(node_info, options)
-        host_name, host_config =
-                   hypervisor.generate_node(node_info, host_config, bhg_version)
+        host_config = hypervisor.generate_node(node_info, host_config, bhg_version)
         config['CONFIG'].deep_merge!(hypervisor.global_config())
 
+        # Merge in any arbitrary key-value host settings. Treat the 'hostname'
+        # setting specially, and don't merge it in as an arbitrary setting.
+        arbitrary_settings = node_info['host_settings']
+        if arbitrary_settings['hostname']
+          host_name = arbitrary_settings['hostname']
+          arbitrary_settings.delete('hostname')
+        end
+        host_config.merge!(arbitrary_settings)
```

Comment:
Rather than having callers create copies of hashes, I've defined the hashes as methods that return a hash rather than a const that is a hash: https://github.com/nwolfe/beaker-hostgenerator/blob/qeng3276/none-hypervisor-and-host-settings/lib/beaker-hostgenerator/data.rb#L56-L63